### PR TITLE
Bump patch version for 3.11.1 release

### PR DESCRIPTION
**Pull request by michaelosthege**
This bumps the patch version to prepare the `3.11.1` release.

Release notes were updated accordingly.

### CODE REVIEW

**Comment by ricardoV94** [comment]
Is the new Theano-pymc version requirement specified in a prior commit?

**Comment by michaelosthege** [comment]
> 
> 
> Is the new Theano-pymc version requirement specified in a prior commit?

That was done in https://github.com/pymc-devs/pymc3/commit/03d7af5b6dd5ad99ab2f3bd8ca7987a744dbef46 about two weeks ago.

The diff is just because I inserted a space that was missing.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
+## PyMC3 3.11.1 (12 February 2021)
 
 ### New Features
 - Automatic imputations now also work with `ndarray` data, not just `pd.Series` or `pd.DataFrame` (see[#4439](https://github.com/pymc-devs/pymc3/pull/4439)).
 - `pymc3.sampling_jax.sample_numpyro_nuts` now returns samples from transformed random variables, rather than from the unconstrained representation (see [#4427](https://github.com/pymc-devs/pymc3/pull/4427)).
 
 ### Maintenance
 - We upgraded to `Theano-PyMC v1.1.2` which [includes bugfixes](https://github.com/pymc-devs/aesara/compare/rel-1.1.0...rel-1.1.2) for warning floods and compiledir locking (see [#4444](https://github.com/pymc-devs/pymc3/pull/4444))
-- `Theano-PyMC v1.1.2` also fixed an important issue in `tt.switch` that affected the behavior of several PyMC distributions, including at least the `Bernoulli` and `TruncatedNormal` (see[#4448](https://github.com/pymc-devs/pymc3/pull/4448))
+- `Theano-PyMC v1.1.2` also fixed an important issue in `tt.switch` that affected the behavior of several PyMC distributions, including at least the `Bernoulli` and `TruncatedNormal` (see [#4448](https://github.com/pymc-devs/pymc3/pull/4448))
```
Comment (twiecki) [inline RELEASE-NOTES.md]: This sounds quite disconcerting without more context. This only affected the bounds, not the inference, right?

Comment (twiecki) [reply RELEASE-NOTES.md]: I.e. the question if people got wrong results.

Comment (ricardoV94) [reply RELEASE-NOTES.md]: It affected inference as well:
1. For the Bernoulli only on the edge case where all observed values were 0 or 1
2. For the TruncatedNormal more often (for observed variables), when mu / sigma were being broadcasted (gradient failed, but not logp). If mu and sigma were not being broadcasted it seems to have been working fine (as seen in the new pymc-examples Truncated Regression).

Comment (twiecki) [reply RELEASE-NOTES.md]: In that case we should make a bigger deal out of this. Maybe we should write a wiki page describing this in more detail and link it here. But if people potentially got wrong answers and need to redo analyses we owe it to our users to be fully transparent about this and warn them properly.

Comment (ricardoV94) [reply RELEASE-NOTES.md]: I can do it on Monday. If someone wants to write up in the meanwhile, here is the illustrating example for the TruncatedNormal:

```python
import arviz as az
import numpy as np
import pymc3 as pm


np.random.seed(2021)
x = np.random.normal(0, 2, size=5000)
obs = x[(x >= -1) & (x <= 2)]

# Problems with NUTS (but not Metropolis, since it is a gradient bug) when mu has to be broadcasted
with pm.Model() as m1:
    mu = pm.Normal('mu', 0, 5)
    x = pm.TruncatedNormal('x', mu=mu, sigma=2, lower=-1, upper=2, observed=obs, shape=len(obs))
    trace1 = pm.sample()

print(az.summary(trace1))  # convergence stats look bad
#       mean    sd  hdi_3%  hdi_97%  ...  ess_sd  ess_bulk  ess_tail  r_hat
# mu -0.347  0.55  -0.938     0.27  ...     2.0       3.0      13.0   2.27


# Fine when mu does not need to be broadcasted
with pm.Model() as m2:
    mu = pm.Normal('mu', 0, 5)
    x = pm.TruncatedNormal('x', mu=mu+np.zeros_like(obs), sigma=2, lower=-1, upper=2, observed=obs, shape=len(obs))
    trace2 = pm.sample()

print(az.summary(trace2))  # converge stats look fine
#      mean    sd  hdi_3%  hdi_97%  ...  ess_sd  ess_bulk  ess_tail  r_hat
# mu  0.112  0.094  -0.058    0.296  ...   896.0     938.0    1351.0    1.0

```

I think things were working fine for unobserved TruncatedNormal variables.
~~I did not test whether having the sigma broadcasted also fixes it~~. Broadcasting `sigma` does not solve it. Things are fine if `sigma` is a random variable, but if it's a fixed value, as above, then `mu` needs to be explicitly broadcasted.

This should be fine after theano-pymc 1.1.2.

Comment (michaelosthege) [reply RELEASE-NOTES.md]: Agree. @ricardoV94 can you give me a more precise description of the issue to replace the current one in the release notes?

We can also link to an issue/PR/wiki page even if the writeup is not done yet.
Let's get the fix released!

Comment (ricardoV94) [reply RELEASE-NOTES.md]: `Theano-PyMC v1.1.2` also fixed an important issue in `tt.switch` that affected the behavior of several PyMC distributions, including at least the following cases (see [#4448](https://github.com/pymc-devs/pymc3/pull/4448)):
1.  `Bernoulli` when all the observed values were the same (e.g., `[0, 0, 0, 0, 0]`).
2.  `TruncatedNormal` when `sigma` was constant and `mu` was being automatically broadcasted to match the shape of observations.

**Review by ricardoV94 — APPROVED** [review]
